File: VB/Q143451/MyPivotGridControl.vb
Imports DevExpress.XtraPivotGrid
Imports System.Drawing

Namespace DXSample
	Public Class MyPivotGridControl
		Inherits PivotGridControl

		Public Sub New()
			MyBase.New()
		End Sub
'INSTANT VB NOTE: The variable synchronizedPivot was renamed since Visual Basic does not allow variables and other class members to have the same name:
		Private synchronizedPivot_Renamed As MyPivotGridControl
		Public Property SynchronizedPivot() As MyPivotGridControl
			Get
				Return synchronizedPivot_Renamed
			End Get
			Set(ByVal value As MyPivotGridControl)
				synchronizedPivot_Renamed = value
				If synchronizedPivot_Renamed.SynchronizedPivot Is Nothing Then
					synchronizedPivot_Renamed.SynchronizedPivot = Me
					synchronizedPivot_Renamed.Size = Size
				End If
			End Set
		End Property
		Protected Overrides Sub ScrollBarsValueChanged(ByVal newValue As Point)
			MyBase.ScrollBarsValueChanged(newValue)
			synchronizedPivot_Renamed.ScrollBarsValueChangedInternal(newValue)
		End Sub
		Private Sub ScrollBarsValueChangedInternal(ByVal newValue As Point)
			MyBase.ScrollBarsValueChanged(newValue)
		End Sub
	End Class
End Namespace

File: CS/Q143451/MyPivotGridControl.cs
using DevExpress.XtraPivotGrid;
using System.Drawing;

namespace DXSample
{
    public class MyPivotGridControl : PivotGridControl {
        public MyPivotGridControl() : base() { }
        private MyPivotGridControl synchronizedPivot;
        public MyPivotGridControl SynchronizedPivot {
            get { return synchronizedPivot; }
            set {
                synchronizedPivot = value;
                if (synchronizedPivot.SynchronizedPivot == null) {
                    synchronizedPivot.SynchronizedPivot = this;
                    synchronizedPivot.Size = Size;
                }
            }
        }
        protected override void ScrollBarsValueChanged(Point newValue) {
            base.ScrollBarsValueChanged(newValue);
            synchronizedPivot.ScrollBarsValueChangedInternal(newValue);
        }
        private void ScrollBarsValueChangedInternal(Point newValue) {
            base.ScrollBarsValueChanged(newValue);
        }
    }
}